src/components/WeeklyForecast.svelte

```diff
@@ -0,0 +1,153 @@
+  import weatherCodes from '../assets/weather-interpretation-code-description.json';
+  import { t } from 'svelte-i18n';
+
+  export let latitude;
+  export let longitude;
+
+  const weatherData = {
+    isLoading: false,
+    hasError: false,
+  };
+  let forecastData = [];
+  const routeBase = '/details';
+
+  function formatDate(dateString) {
+    return moment(dateString)
+      .format('dddd DD MMM');
+  }
+
+  async function fetchWeatherData(latitude, longitude) {
+    weatherData.isLoading = true;
+    weatherData.hasError = false;
+
+      const response = await weatherService.getWeeklyForecast(latitude, longitude);
+      if (!response) {
+        weatherData.isLoading = false;
+        weatherData.hasError = true;
+
+        return;
+      }
+
+      const dailyData = response.daily;
+
+      forecastData = dailyData.time?.map((time, i) => ({
+        date: formatDate(time),
+        rawDate: time,
+        maxTemp: dailyData.temperature_2m_max[i],
+        minTemp: dailyData.temperature_2m_min[i],
+        precipitation: dailyData.precipitation_sum[i],
+        windSpeed: dailyData.wind_speed_10m_max[i],
+        weatherCode: dailyData.weather_code[i],
+      }));
+
+      weatherData.isLoading = false;
+  }
+
+  $: if (latitude && longitude) {
+    fetchWeatherData(latitude, longitude);
+  }
+</script>
+
+<div class="p-5 md:p-10">
+  {#if weatherData.isLoading}
+    <div>{$t('loading')}</div>
+  {:else if weatherData.hasError}
+    <div>{$t('errorLoadingData')}</div>
+  {:else}
+    <div class="block md:hidden">
+      <div class="grid grid-cols-1 gap-4">
+        {#each forecastData as day}
+          <div class="bg-white shadow-md rounded-md p-4 flex flex-col items-center">
+            <div class="text-center font-bold">{day.date}</div>
+            {#if day.weatherCode !== null && weatherCodes[day.weatherCode]?.['day']?.image}
+              <img
+                class="w-16 h-16 object-cover"
+                src={weatherCodes[day.weatherCode]['day'].image}
+                alt="weather"
+              />
+            {/if}
+            <div class="text-lg">
+              <span class={`text-${day.maxTemp >= 0 ? 'red' : 'blue'}-500`}>{day.maxTemp}°C</span>
+              /
+              <span class={`text-${day.minTemp >= 0 ? 'red' : 'blue'}-500`}>{day.minTemp}°C</span>
+            </div>
+            <div class="text-sm">
+              <span class="text-blue-500"
+                >{day.precipitation} {$t('weeklyForecast.millimeters')}</span
+              >
+              |
+              <span>{day.windSpeed} {$t('weeklyForecast.kilometersPerHour')}</span>
+            </div>
+            <a
+              href={`${routeBase}/${day.rawDate}`}
+              use:link
+              class="text-blue-500 hover:underline mt-2"
+            >
+              {$t('weeklyForecast.openHourlyForecast')}
+            </a>
+          </div>
+        {/each}
+      </div>
+    </div>
+
+    <div class="hidden md:block">
```

Comment:
maybe better to use `lg:block` to avoid horizontal scrolling
![image](https://github.com/user-attachments/assets/bb68f653-8e95-4599-859e-d21a99fc4290)

Reply:
Maybe. I'll try your suggestion)